utils/treesitter/test.cpp

```diff
@@ -0,0 +1,64 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <tree_sitter/api.h>
+#include <tree_sitter/parser.h>
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+extern "C" {
+TSLanguage* tree_sitter_carbon();
+}
+
+// Reads a file to string.
+static auto ReadFile(std::filesystem::path path) -> std::string {
+  std::ifstream file(path);
+  std::stringstream buffer;
+  buffer << file.rdbuf();
+  file.close();
+  return buffer.str();
+}
+
+auto main(int argc, char** argv) -> int {
```

Comment:
This should probably be using gtest/file_test to make better use of standard infrastructure. Please at least add a TODO.